src/lib/browser/BrowserPage.ts

```
@@ -668,6 +670,224 @@ export class BrowserPage {
+      if (lower.includes('.pdf')) return true;
+      if (lower.includes('/pdf/')) return true;  // Common pattern (e.g., arXiv)
+      if (lower.startsWith('chrome-extension://') && (lower.includes('pdf-viewer') || lower.includes('src='))) return true;
+      // Best-effort HEAD probe for content-type
+      try {
+        const controller = new AbortController();
+        const timeoutId = setTimeout(() => controller.abort(), 3000);
+        const resp = await fetch(currentUrl, { method: 'HEAD', credentials: 'include', signal: controller.signal });
+        clearTimeout(timeoutId);
+        const ct = resp.headers.get('content-type') || '';
+        if (ct.toLowerCase().includes('application/pdf')) return true;
+      } catch {}
```

ESLint

[error] Empty block statement.

(no-empty)